docs/source/api-main-setup.rst

```diff
@@ -0,0 +1,167 @@
+
+
+Install Postgres
+----------------
+
+**For Linux Users**
+
+Before you get started, update your system with this command:
+
+``sudo apt-get update``
+
+Make sure you have python3 and pip installed:
+
+``sudo apt install python3``
+
+``sudo apt install python-pip``
+
+Now, begin to install Postgres:
+
+``sudo apt-get install PostgreSQL PostgreSQL-contrib``
+
+Run the following command. It will take inside the Postgres world.
+
+``sudo -u postgres psql postgres``
+
+Every command now should start with postgres=#
+
+In the postgres world, run the following commands:
+
+``#\password postgres``
+
+You should be prompted to enter a new password. Don’t type anything, just hit enter twice. This should clear the password.
+
+postgres=# ``create database lookit``
+
+postgres=# ``grant all privileges on database lookit to postgres``
+
+If at this point you still do not have access to the lookit database, run the following commands:
+
+``sudo vi /etc/postgresql/10/main/pg_hba.conf``
+
+``pg_ctl reload``
+
+A long document with # on the leftmost side of almost every line should open up. Scroll to the bottom. There will be a few lines that don’t start with #. They might be a different color and will start with either local or host. The last word in each of those lines should be trust. If it is not, switch into editing mode ( hit esc then type i and hit enter ) and change them to say trust. Then, save the file ( hit esc and then type :x before hitting enter ). You should now have access.
+
+
+
+
+
+Install RabbitMQ
+----------------
+
+**For Linux Users**
+
+First, run the following command:
+
+``sudo apt install rabbitmq-server``
+
+Now that rabbitmq server is installed, create an admin user with the following commands:
+
+``sudo rabbitmqctl add_user admin password``
+
+``sudo rabbitmqctl set_user_tags admin administrator``
+
+``sudo rabbitmqctl set_permissions -p / admin ".*" ".*" ".*"``
+
+Make sure that the server is up and running:
+
+``sudo systemctl stop rabbitmq-server.service``
+
+``sudo systemctl start rabbitmq-server.service``
+
+``sudo systemctl enable rabbitmq-server.service``
+
+If you are having problems creating a user or getting the server running, try the following commands:
+
+``sudo rabbitmq-plugins enable rabbitmq_management``
+
+``sudo chown -R rabbitmq:rabbitmq /var/lib/rabbitmq/``
+
+``sudo rabbitmqadmin declare queue --vhost=/ name=email``
+
+``sudo rabbitmqadmin declare queue --vhost=/ name=builds``
+
+``sudo rabbitmqadmin list queues``
+
+When you run the last command, you should see something like this: ::
+
+    +--------+----------+
+    | name  | messages  |
+    +--------+----------+
+    | builds|     0     |
+    | email |     0	|
+    +--------+----------+
+
+
+Install Ngrok
+-------------
+
+**For Linux Users**
+
+To install, run this command:
+
+``sudo snap install ngrok``
+
+To connect to your local host run this command:
+
+``ngrok http “[https://localhost:8000](https://localhost:8000)"``
+
+
```

Comment:
Is this all one needs to do to have ngrok just work? Do we need to get the address and edit on pipe where the webhooks for dev are going anymore?

Reply:
More generally I guess my question is that if someone new installs these things, and then goes to the lookit-api installation instructions, are they all going to play nicely together by magic? or are there other steps that would be needed?

Reply:
I'm not sure, I think it worked for me when I installed it, but I'll check with Rico